### PR TITLE
Upgrade to async-blobstore snapshot with azure v12

### PR DESCRIPTION
**Pull request by rintcius**
Part of ch11156

### CODE REVIEW

**Comment by rintcius** [comment]
Marked as `breaking` though technically it may not be. Don't want this to be auto-merged in slamx though

**Comment by rintcius** [comment]
Published snapshot `24.0.5-7fddcb2`

File: core/src/test/scala/quasar/physical/blobstore/BlobstoreDatasourceSpec.scala [hunk 1]
```diff
@@ -48,25 +45,25 @@ abstract class BlobstoreDatasourceSpec[F[_]: Effect] extends EffectfulQSpec[F] {
 
   "prefixedChildPaths" >> {
 
-    "list nested children" >>* {
+    "list nested children" in IO {
```
Comment (wemrysi) [inline core/src/test/scala/quasar/physical/blobstore/BlobstoreDatasourceSpec.scala:48]: You don't need to wrap all the examples in `IO`, `cats-effect-testing` adds support for treating `IO` values as results (automatically running them with a timeout), it doesn't mandate that all examples be in `IO`. I would have expected all the examples to compile as-is, once you replaced `>>*` with `>>`.

**Comment by alissapajer** [comment]
`24.0.7-c1bcde3`

**Pull request merged**